src/library_pthread.js

```diff
@@ -983,6 +983,27 @@ var LibraryPThread = {
     PThread.currentProxiedOperationCallerThread = callingThread;
     var rtn = func(...proxiedJSCallArgs);
     PThread.currentProxiedOperationCallerThread = 0;
+    if (promiseCtx) {
+#if ASSERTIONS
+      assert(!!rtn.then, 'Return value of proxied function expected to be a Promise but got' + rtn);
+#endif
+      if (!rtn.then) {
+        throw new Error('Return value of proxied function expected to be a Promise but got' + rtn);
+      }
```

Comment:
Need for both that assertion and runtime check here.  I think you can just remove these 3 lines.

Reply:
oops. nice catch